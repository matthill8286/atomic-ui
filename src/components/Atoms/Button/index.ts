export * from './Button'
export * from './Button.interface'
export * from './IconButton'
