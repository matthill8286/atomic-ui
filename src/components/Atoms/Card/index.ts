export * from './CardV2'
export * from './Card.interface'
export { StyledCard } from './Card.styled'
