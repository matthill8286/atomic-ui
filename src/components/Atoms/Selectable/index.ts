export * from './Selectable'
export * from './Selectable.interface'
