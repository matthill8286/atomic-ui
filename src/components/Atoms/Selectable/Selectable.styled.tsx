import { lighten } from 'polished'
import { Card } from '@/components/Atoms/Card'
import { Icon } from '@/components/Atoms/Icon'
import { CopyText } from '@/components/Atoms/Typography/CopyText'
import { styled } from '@/styles/styled'
import {
  SelectableState,
  StyledLabelProps,
  StyledNativeSelectableProps,
  StyledSelectedCheckMarkProps,
} from './Selectable.interface'

const getSelectableSize = (selectableSize: string) => (selectableSize === 'large' ? '24px' : '20px')

const getSelectableFillingColor = (state: SelectableState, theme) =>
  state === 'disabled' ? theme.color.grey4 : theme.color.primary

export const StyledIcon = styled(Icon)<StyledSelectedCheckMarkProps>`
  display: none;
  position: absolute;
  top: 0;
  left: 0;
  width: ${({ selectableSize }) => getSelectableSize(selectableSize)};
  height: ${({ selectableSize }) => getSelectableSize(selectableSize)};
  svg {
    fill: ${({ theme, state }) => getSelectableFillingColor(state, theme)};
    height: 100%;
  }
`
export const StyledRadioMark = styled.span<StyledSelectedCheckMarkProps>`
  width: ${({ selectableSize }) => getSelectableSize(selectableSize)};
  height: ${({ selectableSize }) => getSelectableSize(selectableSize)};
  position: absolute;
  display: none;
  &:before {
    content: '';
    background: ${({ state, theme }) => getSelectableFillingColor(state, theme)};
    width: ${({ selectableSize }) => (selectableSize === 'large' ? '12px' : '10px')};
    height: ${({ selectableSize }) => (selectableSize === 'large' ? '12px' : '10px')};
    border-radius: 50%;
  }
`

export const StyledLabel = styled.label<StyledLabelProps>`
  cursor: pointer;
  position: relative;
  top: 0;
  ${({
    state,
    theme: {
      polished: { inactive },
    },
  }) =>
    state === 'disabled' &&
    `
    pointer-events: none;
    opacity: ${inactive};
  `}
  display: flex;
  height: 100%;
  align-items: center;
  &:before {
    content: '';
    top: 0;
    position: absolute;
    width: ${({ selectableSize }) => getSelectableSize(selectableSize)};
    height: ${({ selectableSize }) => getSelectableSize(selectableSize)};
    border-radius: ${({ theme, type }) =>
      type === 'radio' ? '50%' : theme.dimension.borderRadius1};
    border: solid 1px
      ${({ theme, state }) => (state === 'error' ? theme.color.error : theme.color.grey4)};
    background-color: ${({ theme, state }) =>
      state === 'disabled'
        ? lighten(theme.polished.inactive, theme.color.grey3)
        : theme.color.white};
    box-sizing: border-box;
  }
`
export const StyledFocusedIndicator = styled.span`
  display: inline-block;
`

export const StyledNativeSelectable = styled.input<StyledNativeSelectableProps>`
  width: ${({ selectableSize }) => getSelectableSize(selectableSize)};
  height: ${({ selectableSize }) => getSelectableSize(selectableSize)};
  position: relative;
  z-index: -1;
  display: flex;
  align-items: center;
  justify-content: center;
  opacity: 0;
  flex-shrink: 0;
  ${({ type }) =>
    type === 'checkbox' &&
    `&:checked + ${StyledIcon} {
      display: inline-block;
      text-align: center;
  }`};

  ${({ type }) =>
    type === 'radio' &&
    `&:checked + ${StyledRadioMark} {
      display: flex;
      align-items: center;
      justify-content: center;
      position: absolute;
      left: 0;
      top: 0;
  }`};

  &:focus ~ ${StyledFocusedIndicator} {
    width: ${({ selectableSize }) => getSelectableSize(selectableSize)};
    height: ${({ selectableSize }) => getSelectableSize(selectableSize)};
    box-shadow: ${({ theme }) => theme.dimension.elevationLevel1};
    position: absolute;
    top: 0;
    z-index: -1;
    border-radius: ${({ type }) => (type === 'checkbox' ? '4px' : '50%')};
    left: 0;
  }
`

export const StyledCopyText = styled(CopyText).attrs(() => ({ tag: 'div' }))`
  width: 100%;
`

export const StyledSelectableContentWrapper = styled.div`
  width: 100%;
`

export const StyledSelectableButton = styled.input`
  display: none;
`

export const StyledSelectableContent = styled.div`
  display: flex;
  align-items: center;
`

export const StyledSelectableButtonContentWrapper = styled(Card)`
  & > div {
    display: flex;
    justify-content: center;
    height: 40px;
  }
`

export const StyledButtonLabel = styled.label`
  cursor: pointer;
`

export const StyledButtonText = styled(CopyText).attrs(() => ({ tag: 'div' }))`
  line-height: ${({ theme }) => theme.font.lineHeight.sm};
  margin-left: ${({ theme }) => theme.spacing.base.xxs};
  width: 100%;
`
