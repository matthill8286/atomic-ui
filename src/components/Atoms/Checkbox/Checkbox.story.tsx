import { action } from '@storybook/addon-actions'
import { boolean, select, text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { SelectableState } from '@/components/Atoms/Selectable/Selectable.interface'
import { Checkbox } from './Checkbox'

const selectableStates: { [key in SelectableState]: SelectableState } = {
  idle: 'idle',
  error: 'error',
  disabled: 'disabled',
}

storiesOf('Design System/Atoms/Checkbox', module).add('Checkbox', () => {
  const knobs = () => {
    return {
      isChecked: boolean('Checked', false),
      state: select('State', selectableStates, 'idle'),
      errorMessage: text('Errormessage', 'Ops something went wrong!'),
      selectableSize: select('Selectable size', { Large: 'large', Small: 'small' }, 'large'),
      label: text('Label', 'Label name'),
      value: text('Value', 'Some value'),
    }
  }
  return (
    <Checkbox
      name="checkbox"
      selectableId="selectableId"
      {...knobs()}
      onChangeValue={action('onchangevalue')}
    />
  )
})
