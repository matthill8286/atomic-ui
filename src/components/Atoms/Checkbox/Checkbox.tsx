import * as React from 'react'
import { Selectable } from '@/components/Atoms/Selectable'
import { CheckboxProps } from '@/components/Atoms/Selectable/Selectable.interface'

export const Checkbox: React.FC<CheckboxProps> = ({ ...props }) => {
  return <Selectable type="checkbox" {...props} data-test={'checkbox'} />
}
