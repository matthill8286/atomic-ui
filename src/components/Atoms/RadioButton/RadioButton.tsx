import * as React from 'react'
import { Selectable } from '@/components/Atoms/Selectable'
import { RadioButtonProps } from '@/components/Atoms/Selectable/Selectable.interface'

export const RadioButton: React.FC<RadioButtonProps> = ({ ...props }) => {
  return <Selectable type="radio" {...props} />
}
