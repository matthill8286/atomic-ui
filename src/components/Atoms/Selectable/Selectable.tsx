import * as React from 'react'
import { Icon } from '../Icon'
import { CopyText } from '../Typography'
import { SelectableProps } from './Selectable.interface'
import {
  StyledButtonLabel,
  StyledButtonText,
  StyledCopyText,
  StyledFocusedIndicator,
  StyledIcon,
  StyledLabel,
  StyledNativeSelectable,
  StyledRadioMark,
  StyledSelectableButton,
  StyledSelectableButtonContentWrapper,
  StyledSelectableContent,
  StyledSelectableContentWrapper,
} from './Selectable.styled'
import { IconDone } from '@matthill8286/atomic-icon-library'

export const Selectable: React.FC<SelectableProps> = ({
  children,
  displayType,
  errorMessage,
  icon,
  isChecked,
  label,
  selectableId,
  state = 'idle',
  onChangeValue,
  selectableSize = 'large',
  type,
  value = '',
  name,
  inputRef,
  ...otherProps
}) => {
  const onInputChange = event => {
    onChangeValue?.(event.target.checked, event.target.value)
  }

  const isCheckedValue = isChecked && { checked: isChecked }
  const showError = state === 'error' && errorMessage

  const SelectableRadioOrCheckbox = (
    <StyledLabel
      key={selectableId + isChecked}
      selectableSize={selectableSize}
      state={state}
      type={type}
      htmlFor={selectableId}
      {...otherProps}>
      <StyledNativeSelectable
        id={selectableId}
        {...isCheckedValue}
        type={type}
        name={name}
        value={value}
        state={state}
        selectableSize={selectableSize}
        onChange={e => onInputChange(e)}
        ref={inputRef}
      />
      {type === 'checkbox' && (
        <StyledIcon selectableSize={selectableSize} state={state}>
          <IconDone />
        </StyledIcon>
      )}
      {type === 'radio' && <StyledRadioMark selectableSize={selectableSize} state={state} />}
      <StyledSelectableContentWrapper>
        <StyledCopyText
          padding="0 0 0 xs"
          fontSize={selectableSize === 'large' ? 'sm' : 'xs'}
          lineHeight={selectableSize === 'large' ? 'sm' : 'xs'}>
          {children || label}
        </StyledCopyText>
        {showError && (
          <CopyText
            tag="div"
            color="error"
            fontSize="xxs"
            margin="xs 0 0 0"
            padding="0 0 0 xs"
            data-test={`${name}__error`}>
            {errorMessage}
          </CopyText>
        )}
      </StyledSelectableContentWrapper>
      <StyledFocusedIndicator />
    </StyledLabel>
  )

  const SelectableButton = (
    <StyledButtonLabel key={selectableId + isChecked} htmlFor={selectableId}>
      <StyledSelectableButton
        id={selectableId}
        {...isCheckedValue}
        type={type}
        name={name}
        value={value}
        onChange={e => onInputChange(e)}
        ref={inputRef}
      />
      <StyledSelectableButtonContentWrapper
        elevation={isChecked ? 1 : 0}
        shape="rounded-small"
        surface={isChecked ? 'selected' : 'white'}
        padding={{ top: 'xs', right: 'sm', bottom: 'xs', left: 'sm' }}>
        <StyledSelectableContent>
          <Icon
            height={selectableSize === 'large' ? 14 : 12}
            width={selectableSize === 'large' ? 14 : 12}>
            {icon}
          </Icon>
          <StyledButtonText
            bold={isChecked}
            color={isChecked ? 'black' : 'grey4'}
            fontSize={selectableSize === 'large' ? 'sm' : 'xs'}
            lineHeight={selectableSize === 'large' ? 'sm' : 'xs'}>
            {children || label}
          </StyledButtonText>
        </StyledSelectableContent>
      </StyledSelectableButtonContentWrapper>
      <StyledFocusedIndicator />
    </StyledButtonLabel>
  )

  return displayType === 'button' ? SelectableButton : SelectableRadioOrCheckbox
}
