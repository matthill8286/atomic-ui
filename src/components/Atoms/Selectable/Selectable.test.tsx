import React from 'react'
import { Checkbox } from '@/components/Atoms/Checkbox'
import { RadioButton } from '@/components/Atoms/RadioButton'
import { RadioButtonProps } from '@/components/Atoms/Selectable/Selectable.interface'
import { renderWithTheme } from '@/testRenderer'

const mockProps: RadioButtonProps = {
  label: '',
  value: '',
  name: '',
  onChangeValue: jest.fn(),
  selectableId: '',
  isChecked: false,
  state: 'idle',
}

describe('renders Selectables', () => {
  it('renders checkbox correctly', () => {
    const tree = renderWithTheme(<Checkbox {...mockProps} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders radio button correctly', () => {
    const tree = renderWithTheme(<RadioButton {...mockProps} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders radio button group correctly', () => {
    const tree = renderWithTheme(
      <>
        <RadioButton {...mockProps} selectableId="radio1" name="radios" />
        <RadioButton {...mockProps} selectableId="radio2" name="radios" />
      </>
    )
    expect(tree).toMatchSnapshot()
  })
})
