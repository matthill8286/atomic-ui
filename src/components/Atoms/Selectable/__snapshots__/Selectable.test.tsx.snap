// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Selectables renders checkbox correctly 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 svg {
  width: 25px;
  height: 25px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: currentColor;
}

.c3 svg g,
.c3 svg p {
  fill: currentColor;
}

.c6 {
  padding: 0 0 0 8px;
  color: #666666;
  font-size: 16px;
  line-height: 24px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c4 {
  display: none;
  position: absolute;
  top: 0;
  left: 0;
  width: 24px;
  height: 24px;
}

.c4 svg {
  fill: #00C8DC;
  height: 100%;
}

.c0 {
  cursor: pointer;
  position: relative;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:before {
  content: '';
  top: 0;
  position: absolute;
  width: 24px;
  height: 24px;
  border-radius: 4px;
  border: solid 1px #999999;
  background-color: #FFFFFF;
  box-sizing: border-box;
}

.c9 {
  display: inline-block;
}

.c1 {
  width: 24px;
  height: 24px;
  position: relative;
  z-index: -1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c1:checked + .c2 {
  display: inline-block;
  text-align: center;
}

.c1:focus ~ .c8 {
  width: 24px;
  height: 24px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.07),0 2px 1px -1px rgba(0,0,0,0.06),0 1px 3px 0 rgba(0,0,0,0.10);
  position: absolute;
  top: 0;
  z-index: -1;
  border-radius: 4px;
  left: 0;
}

.c7 {
  width: 100%;
}

.c5 {
  width: 100%;
}

<label
  className="c0"
  data-test="checkbox"
  htmlFor=""
  type="checkbox"
>
  <input
    className="c1"
    id=""
    name=""
    onChange={[Function]}
    type="checkbox"
    value=""
  />
  <div
    className="c2 c3 c4"
  >
    <svg
      height={32}
      viewBox="0 0 24 24"
      width={32}
    />
  </div>
  <div
    className="c5"
  >
    <div
      className="c6 c7"
      color="grey5"
      fontSize="sm"
    >
      
    </div>
  </div>
  <span
    className="c8 c9"
  />
</label>
`;

exports[`renders Selectables renders radio button correctly 1`] = `
.c5 {
  padding: 0 0 0 8px;
  color: #666666;
  font-size: 16px;
  line-height: 24px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3 {
  width: 24px;
  height: 24px;
  position: absolute;
  display: none;
}

.c3:before {
  content: '';
  background: #00C8DC;
  width: 12px;
  height: 12px;
  border-radius: 50%;
}

.c0 {
  cursor: pointer;
  position: relative;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:before {
  content: '';
  top: 0;
  position: absolute;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  border: solid 1px #999999;
  background-color: #FFFFFF;
  box-sizing: border-box;
}

.c8 {
  display: inline-block;
}

.c1 {
  width: 24px;
  height: 24px;
  position: relative;
  z-index: -1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c1:checked + .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  left: 0;
  top: 0;
}

.c1:focus ~ .c7 {
  width: 24px;
  height: 24px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.07),0 2px 1px -1px rgba(0,0,0,0.06),0 1px 3px 0 rgba(0,0,0,0.10);
  position: absolute;
  top: 0;
  z-index: -1;
  border-radius: 50%;
  left: 0;
}

.c6 {
  width: 100%;
}

.c4 {
  width: 100%;
}

<label
  className="c0"
  htmlFor=""
  type="radio"
>
  <input
    className="c1"
    id=""
    name=""
    onChange={[Function]}
    type="radio"
    value=""
  />
  <span
    className="c2 c3"
  />
  <div
    className="c4"
  >
    <div
      className="c5 c6"
      color="grey5"
      fontSize="sm"
    >
      
    </div>
  </div>
  <span
    className="c7 c8"
  />
</label>
`;

exports[`renders Selectables renders radio button group correctly 1`] = `
Array [
  .c5 {
  padding: 0 0 0 8px;
  color: #666666;
  font-size: 16px;
  line-height: 24px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3 {
  width: 24px;
  height: 24px;
  position: absolute;
  display: none;
}

.c3:before {
  content: '';
  background: #00C8DC;
  width: 12px;
  height: 12px;
  border-radius: 50%;
}

.c0 {
  cursor: pointer;
  position: relative;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:before {
  content: '';
  top: 0;
  position: absolute;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  border: solid 1px #999999;
  background-color: #FFFFFF;
  box-sizing: border-box;
}

.c8 {
  display: inline-block;
}

.c1 {
  width: 24px;
  height: 24px;
  position: relative;
  z-index: -1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c1:checked + .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  left: 0;
  top: 0;
}

.c1:focus ~ .c7 {
  width: 24px;
  height: 24px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.07),0 2px 1px -1px rgba(0,0,0,0.06),0 1px 3px 0 rgba(0,0,0,0.10);
  position: absolute;
  top: 0;
  z-index: -1;
  border-radius: 50%;
  left: 0;
}

.c6 {
  width: 100%;
}

.c4 {
  width: 100%;
}

<label
    className="c0"
    htmlFor="radio1"
    type="radio"
  >
    <input
      className="c1"
      id="radio1"
      name="radios"
      onChange={[Function]}
      type="radio"
      value=""
    />
    <span
      className="c2 c3"
    />
    <div
      className="c4"
    >
      <div
        className="c5 c6"
        color="grey5"
        fontSize="sm"
      >
        
      </div>
    </div>
    <span
      className="c7 c8"
    />
  </label>,
  .c5 {
  padding: 0 0 0 8px;
  color: #666666;
  font-size: 16px;
  line-height: 24px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3 {
  width: 24px;
  height: 24px;
  position: absolute;
  display: none;
}

.c3:before {
  content: '';
  background: #00C8DC;
  width: 12px;
  height: 12px;
  border-radius: 50%;
}

.c0 {
  cursor: pointer;
  position: relative;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:before {
  content: '';
  top: 0;
  position: absolute;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  border: solid 1px #999999;
  background-color: #FFFFFF;
  box-sizing: border-box;
}

.c8 {
  display: inline-block;
}

.c1 {
  width: 24px;
  height: 24px;
  position: relative;
  z-index: -1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c1:checked + .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  left: 0;
  top: 0;
}

.c1:focus ~ .c7 {
  width: 24px;
  height: 24px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.07),0 2px 1px -1px rgba(0,0,0,0.06),0 1px 3px 0 rgba(0,0,0,0.10);
  position: absolute;
  top: 0;
  z-index: -1;
  border-radius: 50%;
  left: 0;
}

.c6 {
  width: 100%;
}

.c4 {
  width: 100%;
}

<label
    className="c0"
    htmlFor="radio2"
    type="radio"
  >
    <input
      className="c1"
      id="radio2"
      name="radios"
      onChange={[Function]}
      type="radio"
      value=""
    />
    <span
      className="c2 c3"
    />
    <div
      className="c4"
    >
      <div
        className="c5 c6"
        color="grey5"
        fontSize="sm"
      >
        
      </div>
    </div>
    <span
      className="c7 c8"
    />
  </label>,
]
`;
